src/document/generator.ml

```diff
@@ -1783,26 +1799,152 @@ module Make (Syntax : SYNTAX) = struct
       in
       List.map f t
 
-    let compilation_unit (t : Odoc_model.Lang.Compilation_unit.t) =
+    module Hierarchy = struct
```

Comment:
The `dir` type is quite complex and probably leaks out of this module. What do you think of abstracting it behind a signature ?

Reply:
Nice!